### ISSUE TITLE
Generate test certificate avec testingCenterOfFacility avec des espaces ne fonctionne pas

### ISSUE THREAD

**Issue by golayp** [admin-ch/CovidCertificate-Api-Gateway-Service#13]
Lors de l'appel de /api/v1/covidcertificate/test
si on met des espaces dans l'attribut testingCentreOrFacility la signature X-Signature n'est pas correctement validée et un message du type: 490 Integrity check failed, the body hash does not match the hash in the header

**Comment by yannikinniger** [comment]
I was not able to reproduce this when applying the regex on the payload before creating the hash. Please make sure you are creating the hash as described here:  https://github.com/admin-ch/CovidCertificate-Apidoc#content-signature

**Comment by golayp** [comment]
Since we have tested successfully all kind of api calls with different datas, I though it was a problem on your side. I actually try api with c# generated client using nswag... maybe it's related to the generated client... but not sure.

Here the successful data sended to genererate a test certificate: 
"testInfo": [
        {
            "typeCode": "LP6464-4",
            "manufacturerCode": "",
            "sampleDateTime": "2021-06-02T06:26:25.720Z",
            "testingCentreOrFacility": "centre",
            "memberStateOfTest": "CH"
        }
    ],

And the unsuccessful data sended:
"testInfo": [
        {
            "typeCode": "LP6464-4",
            "manufacturerCode": "",
            "sampleDateTime": "2021-06-02T06:26:25.720Z",
            "testingCentreOrFacility": "centre test",
            "memberStateOfTest": "CH"
        }
    ],

And the response:
{
    "type": "https://tools.ietf.org/html/rfc7231#section-6.6.1",
    "status": 403,
    "detail": "Error code : 490 Detail : Integrity check failed, the body hash does not match the hash in the header"
}

**Comment by delixfe** [comment]
You have a space in the unsuccessful data: `"centre test"`. I guess that you do not remove all spaces from message before you apply the signature. 
Have a look at the regex from the documentation: `/[\n\r\t ]/gm` 
The list includes the space character. This applies also to JSON values.

**Comment by golayp** [comment]
OK tested with spaces on other values fail too... sorry. I will check on my side.

**Comment by golayp** [comment]
Hi @delixfe , I've tried numerous times but unable to fix this problem.
My payload before signing has no space nore multiline since I have used the provided regex to remove them.
Here my payload:
`{"name":{"familyName":"FedererMartin","givenName":"Roger"},"dateOfBirth":"1950-06-04","language":"fr","otp":"xxx","vaccinationInfo":[{"medicinalProductCode":"EU/1/20/1507","numberOfDoses":2,"totalNumberOfDoses":2,"vaccinationDate":"2021-05-14","countryOfVaccination":"CH"}]}`
This is the string I sign (otp removed for security purpose).

This payload reflect the data:
`{"name":{"familyName":"Federer Martin","givenName":"Roger"},"dateOfBirth":"1950-06-04","language":"fr","otp":"xxx","vaccinationInfo":[{"medicinalProductCode":"EU/1/20/1507","numberOfDoses":2,"totalNumberOfDoses":2,"vaccinationDate":"2021-05-14","countryOfVaccination":"CH"}]}`
Any help would be appreciate
